spras/config.py

```diff
@@ -222,16 +223,43 @@ def process_config(self, raw_config):
                 if cur_params[run_params]:
                     for p in cur_params[run_params]:
                         param_name_list.append(p)
-                        all_runs.append(eval(str(cur_params[run_params][p])))
+                        obj = str(cur_params[run_params][p])
+                        try:
+                            obj = [int(obj)]
+                        except ValueError:
+                            try:
+                                obj = [float(obj)]
+                            except ValueError:
+                                # Handles arrays and special evaluation types
+                                # TODO: do we want to explicitly bar `eval` if we may use untrusted user inputs later?
+                                if obj.startswith(("range", "np.linspace", "np.arange", "np.logspace", "[")):
+                                    obj = eval(obj)
+                                elif obj.lower() == "true":
+                                    obj = [True]
+                                elif obj.lower() == "false":
+                                    obj = [False]
+                                else:
+                                    # Catch-all for strings
+                                    obj = [obj]
+                            if not isinstance(obj, Iterable):
+                                raise ValueError(f"The object `{obj}` in algorithm {alg['name']} at key '{p}' in run '{run_params}' is not iterable!") from None
```

Comment:
I'm not a fan of the massive chain of try-catch statements - is there some parsing method that allows me to safely parse `int`s and `float`s with `Optional`s instead of this?

Reply:
@tristan-f-r  is this something you are planning on doing? Otherwise I was going to approve this PR.